app/main.py

```diff
@@ -1,3 +1,51 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __add__(self, other: Distance) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(km=self.km + other)
+        else:
+            raise TypeError(f"Unsupported operand type(s) for +: 'Distance' and '{type(other).__name__}'")
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __iadd__(self, other: Distance) -> Distance:
```

Comment:
The type annotation for the 'other' parameter in the __iadd__ method should include both 'Distance' and 'Union[int, float]' to reflect that it can accept either a Distance object or a numeric type.